src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -0,0 +1,205 @@
+            Assert.True(QueueExists("MsmqQueueCreatorTests.target1"));
+        }
+
+        [Test]
+        public void Should_setup_permissions()
+        {
+            var testQueueName = "MsmqQueueCreatorTests.permissions";
+
+            var creator = new QueueCreator(true);
+            var bindings = new QueueBindings();
+
+            DeleteQueueIfPresent(testQueueName);
+
+            bindings.BindReceiving(testQueueName);
+
+            // use the network service account since that one won't be in the local admin group
+            creator.CreateQueueIfNecessary(bindings, NetworkServiceAccountName);
+
+            var createdQueue = GetQueue(testQueueName);
+
+            MessageQueueAccessRights? accountAccessRights;
+
+            Assert.True(createdQueue.TryGetPermissions(NetworkServiceAccountName, out accountAccessRights));
+            Assert.True(accountAccessRights.HasValue, "User should have access rights");
+            Assert.AreEqual(MessageQueueAccessRights.WriteMessage | MessageQueueAccessRights.ReceiveMessage, accountAccessRights, "User should have write/read access");
+
+            MessageQueueAccessRights? localAdminAccessRights;
+
+            Assert.True(createdQueue.TryGetPermissions(LocalAdministratorsGroupName, out localAdminAccessRights));
+            Assert.True(localAdminAccessRights.HasValue, "User should have access rights");
+            Assert.AreEqual(MessageQueueAccessRights.FullControl, localAdminAccessRights, "LocalAdmins should have full control");
+        }
+
+        [Test]
+        public void Should_make_queues_transactional_if_requested()
+        {
+            const bool isTransactional = true;
+
+            var testQueueName = "MsmqQueueCreatorTests.txreceiver";
+
+            DeleteQueueIfPresent(testQueueName);
+
+            var creator = new QueueCreator(isTransactional);
+            var bindings = new QueueBindings();
+
+            bindings.BindReceiving(testQueueName);
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+            var queue = GetQueue(testQueueName);
+
+            Assert.AreEqual(isTransactional, queue.Transactional);
+        }
+
+        [Test]
+        public void Should_make_queues_non_transactional_if_requested()
+        {
+            const bool isTransactional = false;
+
+            var testQueueName = "MsmqQueueCreatorTests.txreceiver";
+
+            DeleteQueueIfPresent(testQueueName);
+
+            var creator = new QueueCreator(isTransactional);
+            var bindings = new QueueBindings();
+
+            bindings.BindReceiving(testQueueName);
+
+            creator.CreateQueueIfNecessary(bindings, WindowsIdentity.GetCurrent().Name);
+
+            var queue = GetQueue(testQueueName);
+
+            Assert.AreEqual(isTransactional, queue.Transactional);
+        }
+
+        [Test]
+        public void Msmq_will_add_everyone_and_anonymous_by_default()
```

Comment:
nitpick, but the naming of this test doesn't seem to align with the rest of the tests.
How about something like `Should_give_everyone_access_rights_when_creating_queues`?